xcdat/averager.py

```diff
@@ -1,9 +1,12 @@
 """Averager module that contains functions related to geospatial averaging.
 """
 
-import xarray as xr
+from typing import Dict, Hashable, List, Literal, Optional, Tuple, Union, get_args
+
 import numpy as np
+import xarray as xr
 
+# TODO: Reorder functions by callstack hierarchy
```

Comment:
I plan on reordering the functions by callstack hierarchy.

It will be:
1. `spatial_averaging`
2. `_validate_region_lon_bounds()`
   - `_is_region_lon_in_domain`
   - `_swap_region_lon_axis()`
3. `_get_axis_weights()`
   - `_get_area_weights()` 
   - `_get_uniform_weights()`
   - `_scale_dim_to_region()`
4. ` _combine_axis_weights()`
5. `_averager()`

Reply:
@pochedls I would start with this comment to get a sense of how things are logically grouped.